### PR TITLE
Upgrade to Jasmine 3

### PR DESCRIPTION
**Pull request by illright**
### Description of the Change

This will replace the default Atom test runner powered by the long outdated Jasmine 1.3 with a new custom test runner powered by the latest Jasmine 3.6.

### Benefits

The documentation of Jasmine 1.3 is very poor, so upgrading the version will allow developers to write tests more efficiently with the new Jasmine docs. It also improves the developer experience because Jasmine 3 is more pleasant to work with.

### Possible Drawbacks

Not that I can think of any.

P.S. This is a copy of the pull request made upstream (https://github.com/atom/autocomplete-plus/pull/1103), refer to it for the CI results and discussion

### CODE REVIEW

File: spec/custom-runner.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    jasmineFocused: true,
+    jasmineJson: true,
+    jasminePass: true,
+    jasmineTagged: true,
+    mockClock: true,
+    mockLocalStorage: true,
+    profile: true,
+    set: true,
+    unspy: true
+  },
```
Comment (aminya) [inline spec/custom-runner.js]: There is a shorter version
```suggestion
  specHelper: true,
```

**Review by aminya — APPROVED** [review]
The tests pass in the upstream CI

**Comment by UziTech** [comment]
We should move this to github actions as well.

**Comment by illright** [comment]
@UziTech what do you mean?

**Comment by UziTech** [comment]
Right now ci is not running on this repo. We should set up GitHub Actions to run the tests on PRs and merges. (Basically copy https://github.com/atom-community/ide-python/blob/master/.github/workflows/CI.yml file to this repo)

**Comment by aminya** [comment]
I have enabled Travis and Appveyor, but GitHub actions is always much better

**Comment by UziTech** [comment]
Are we releasing this as a separate package (autocomplete-plus-plus 😁)? or is the goal to merge it back into atom/autocomplete-plus?

**Comment by illright** [comment]
Currently I'm planning to make mergeable changes, I wonder if I'll have enough patience to stick to this plan

**Pull request merged**